hibernate-core/src/main/java/org/hibernate/boot/registry/classloading/internal/AggregatedClassLoader.java

```diff
@@ -196,18 +196,21 @@ protected URL findResource(String name) {
 	@Override
 	protected Class<?> findClass(String name) throws ClassNotFoundException {
 		final Iterator<ClassLoader> clIterator = newClassLoaderIterator();
+		Throwable t = new Throwable();
 		while ( clIterator.hasNext() ) {
 			final ClassLoader classLoader = clIterator.next();
 			try {
 				return classLoader.loadClass( name );
 			}
-			catch (Exception ignore) {
+			catch (Exception ex) {
+				t.addSuppressed( ex );
```

Comment:
```suggestion
				if ( t == null ) {
					t = ex;
				}
				else {
					t.addSuppressed(ex);
				}
```


Alternatively, if you feel like it, you could also copy-paste a util from Hibernate Search:

https://github.com/hibernate/hibernate-search/blob/f2ccc2370cfecaa8f9257f855f03a840e1a0a05e/util/common/src/main/java/org/hibernate/search/util/common/impl/Throwables.java#L53-L64

`org.hibernate.internal.util.ExceptionHelper` seems like a good place for that in Hibernate ORM.

It makes the code slightly more readable:

```suggestion
				t = ExceptionHelper.combine( t, ex );
```

See for example https://github.com/hibernate/hibernate-search/blob/f2ccc2370cfecaa8f9257f855f03a840e1a0a05e/integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/testsupport/util/LuceneTckBackendAccessor.java#L111-L113